video/transcoder/src/transcoder/job/track_parser.rs

```diff
@@ -56,38 +56,63 @@ pub fn track_parser(
                 match b {
                     mp4::DynBox::Moov(moov) => {
                         if moov.traks.len() != 1 {
-                            yield Err(io::Error::new(io::ErrorKind::InvalidData, anyhow!("moov box must have exactly one trak box")));
+                            yield Err(io::Error::new(
+                                io::ErrorKind::InvalidData,
+                                anyhow!("moov box must have exactly one trak box"),
+                            ));
                             return;
                         }
 
                         yield Ok(TrackOut::Moov(moov));
-                    },
+                    }
                     mp4::DynBox::Moof(moof) => {
                         if moof.traf.len() != 1 {
-                            yield Err(io::Error::new(io::ErrorKind::InvalidData, anyhow!("moof box must have exactly one traf box")));
+                            yield Err(io::Error::new(
+                                io::ErrorKind::InvalidData,
+                                anyhow!("moof box must have exactly one traf box"),
+                            ));
                             return;
                         }
 
                         let traf = &moof.traf[0];
-                        let trun = traf.trun.as_ref().ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, anyhow!("traf box must have a trun box")))?;
+                        let trun = traf.trun.as_ref().ok_or_else(|| {
+                            io::Error::new(
+                                io::ErrorKind::InvalidData,
+                                anyhow!("traf box must have a trun box"),
+                            )
+                        })?;
                         let tfhd = &traf.tfhd;
 
                         let samples = trun.samples.iter().enumerate().map(|(idx, sample)| {
                             let mut sample = sample.clone();
                             sample.duration = sample.duration.or(tfhd.default_sample_duration);
                             sample.size = sample.size.or(tfhd.default_sample_size);
-                            sample.flags = Some(sample.flags.or(if idx == 0 { trun.first_sample_flags } else { None }).or(tfhd.default_sample_flags).unwrap_or_default());
-                            sample.composition_time_offset = Some(sample.composition_time_offset.unwrap_or_default());
+                            sample.flags = Some(
+                                sample
+                                    .flags
+                                    .or(if idx == 0 {
+                                        trun.first_sample_flags
+                                    } else {
+                                        None
+                                    })
```

Comment:
This `if` feels kinda iffy on formatting, but I don't see a better way to show intent.